src/Appwrite/Platform/Workers/StatsUsage.php

```diff
@@ -119,24 +244,47 @@ public function __construct()
             ->inject('getProjectDB')
             ->inject('getLogsDB')
             ->inject('register')
+            ->inject('logError')
             ->callback($this->action(...));
 
         $this->lastTriggeredTime = time();
     }
 
+    /**
+     * Validate if a metric is defined in constants
+     * @param string $metricKey
+     * @return bool
+     */
+    protected function isValidMetric(string $metricKey): bool
+    {
+        if (in_array($metricKey, $this->validMetrics)) {
+            return true;
+        }
+
+        foreach ($this->validMetricPatterns as $pattern => $regex) {
```

Comment:
I'm worried about the additional time this takes patter matching for each key, we are already slow in stats usage containers.